Extractor/FrontendExtractor.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace Bazinga\Bundle\JsTranslationBundle\Extractor;
+
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Translation\Extractor\AbstractFileExtractor;
+use Symfony\Component\Translation\MessageCatalogue;
+use Symfony\Component\Translation\Extractor\ExtractorInterface;
+
+final class FrontendExtractor extends AbstractFileExtractor implements ExtractorInterface
+{
+    const FUNCTION_STRING_ARGUMENT_REGEX = '\s?[\'"]([^"\'),]+)[\'"]\s?';
+    const REGEX_DELIMITER = '/';
+
+    private $prefix = '';
+
+    private $filesystem;
+
+    private $extensions;
+
+    private $sequence;
+
+    public function __construct(Filesystem $filesystem, array $extensions, $sequence) {
+        $this->filesystem = $filesystem;
+        $this->extensions = $extensions;
+        $this->sequence = $sequence;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function extract($resource, MessageCatalogue $catalogue)
+    {
+        $files = $this->extractFiles($resource);
+
+        foreach ($files as $file) {
+            $this->parseMessagesFromContent(file_get_contents($file), $catalogue);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setPrefix($prefix)
+    {
+        $this->prefix = $prefix;
+    }
+
+    private function parseMessagesFromContent($fileContent, MessageCatalogue $catalogue)
+    {
+        $messages = $this->getMessagesForSequence($fileContent);
+
+        foreach ($messages as $message) {
+            $catalogue->set($message, $this->prefix.$message);
+        }
+    }
+
+    protected function canBeExtracted($file)
+    {
+        return $this->isFile($file) && in_array(
+            pathinfo($file, PATHINFO_EXTENSION),
+            $this->extensions
+        );
+    }
+
+    protected function isFile($file)
+    {
+        return $this->filesystem->exists($file);
+    }
+
+    protected function extractFromDirectory($directory)
+    {
+        $finder = new Finder();
+
+        $finder->files();
+
+        foreach ($this->extensions as $supportedExtension) {
+            $finder->name(sprintf('*.%s', $supportedExtension));
+        }
+
+        return $finder->in($directory);
+    }
+
+    private function getMessagesForSequence($fileContent)
+    {
+        $argumentsRegex = self::REGEX_DELIMITER
+            .$this->sequence
+            .self::FUNCTION_STRING_ARGUMENT_REGEX
```

Comment:
why making the part about the function call configurable, but hardcoding the string argument regex ? Why not making the whole thing configurable instead ? Different cases might have different regexes for string arguments (thus, your own regex about string argument may not be the best one, as it forbids having any quotes inside the string, even when it is not the delimiting quote)

Reply:
I don't really know. I forked this feature where @jcchavezs left it.

Reply:
I would say that when I first wrote this I saw it convenient to make the keywords `trans` configurable but now I think it is not a problem so I would hardcode all. I don't think having everything configurable is a valid use case.